### PR TITLE
Update go version to 1.17

### LINKED ISSUE #153: Upgrade client-go and controller-runtime libraries
In order to move to latest libraries, go version needs to be updated as well to 1.17.

- controller-runtime [changes](https://github.com/kubernetes-sigs/controller-runtime/releases/v0.11.0)

### PR DESCRIPTION
**Pull request by ahmadateya**
this PR should fix #153

I have updated the go.mod file with 1.17 and removed the vendor directory to make sure it will use the new version.

running `go mod tidy` and `go mod vendor` updates the go.mod file with some new packages and edits other packages which mad some updates in the vendor directory, and running `make install` gives some warning about the deprecation of `go get` command but it works.

Now I tested the edits with `make install` and `make uninstall` commands but I am not sure if this upgrade may cause another issue because I am new to the flotta-operator and the project, So, How to test this upgrade and make sure it's not breaking something?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,9 @@
 When the CRDs are present in the cluster, the operator can either be run outside or inside the cluster.
 
 ## Prerequisites/Dependencies
+### Go
+The operator requires Go `1.16`, Go `1.18` is not supported yet.
```
Comment (masayag) [inline README.md:10]: With this change, let's update it to `1.17`

**Comment by masayag** [comment]
Thank you for the PR, @ahmadateya 

Please go over https://tip.golang.org/doc/go1.17 and see recommendation that we'd like to apply (e.g. switch from `go get` to `go install`, run gofmt over the project, verify `go vet`)

**Comment by arielireni** [comment]
Thank you for your PR,
Please update the go version [here](https://github.com/project-flotta/flotta-operator/blob/066b8f6c2f527f01ff481b61b8fb8f519b0ff1be/.github/workflows/kind.yaml) and [here](https://github.com/project-flotta/flotta-operator/blob/main/Dockerfile) too :)

**Comment by ahmadateya** [comment]
Great thank you @masayag and @arielireni for your comments I will work on them as soon as possible

**Comment by ahmadateya** [comment]
I am sorry I can't work on this PR until the end of April due to some exams.
I am mentioning this to not interrupt your workflow, I will come back in May and continue work on the PR if the issue isn't fixed then

**Comment by masayag** [comment]
> I am sorry I can't work on this PR until the end of April due to some exams. I am mentioning this to not interrupt your workflow, I will come back in May and continue work on the PR if the issue isn't fixed then

Thanks for the heads up. I'm closing the PR and we'll work on it.

**Pull request closed without merging**